x/mint/types/inflations_test.go

```diff
@@ -0,0 +1,181 @@
+package types_test
+
+import (
+	"github.com/cosmos/cosmos-sdk/simapp"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	simtypes "github.com/cosmos/cosmos-sdk/types/simulation"
+	auth "github.com/cosmos/cosmos-sdk/x/auth/types"
+	"github.com/cosmos/cosmos-sdk/x/mint/types"
+	"github.com/stretchr/testify/require"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+	"math"
+	"math/rand"
+	"strconv"
+	"testing"
+)
+
+func resetSupply(app *simapp.SimApp, ctx sdk.Context, initSupply sdk.Coins, curSupply sdk.Coins) {
+	err := app.BankKeeper.MintCoins(ctx, types.ModuleName, initSupply)
+	if err != nil {
+		panic(err)
+	}
+	err = app.BankKeeper.BurnCoins(ctx, types.ModuleName, curSupply)
+	if err != nil {
+		panic(err)
+	}
+}
+
+func getTestingAccounts(r *rand.Rand, n int, ctx sdk.Context, app *simapp.SimApp) []simtypes.Account {
+	accounts := simtypes.RandomAccounts(r, n)
+
+	for _, account := range accounts {
+		acc := app.AccountKeeper.NewAccountWithAddress(ctx, account.Address)
+		app.AccountKeeper.SetAccount(ctx, acc)
+	}
+
+	return accounts
+}
+
+func TestHandleInflations(t *testing.T) {
+	app := simapp.Setup(false)
+	ctx := app.BaseApp.NewContext(false, tmproto.Header{})
+	acc := auth.NewEmptyModuleAccount(types.ModuleName, auth.Minter, auth.Burner)
+	app.AccountKeeper.SetModuleAccount(ctx, acc)
+
+	s := rand.NewSource(1)
+	r := rand.New(s)
+	targetAccounts := getTestingAccounts(r, 3, ctx, app)
+
+	minter := types.DefaultInitialMinter()
+	params := types.DefaultParams()
+
+	app.MintKeeper.SetParams(ctx, params)
+	app.MintKeeper.SetMinter(ctx, minter)
+
+	testDenom := "testDenom"
+	initSupplyAmount, _ := sdk.NewIntFromString("1000000000000000000")
+	initSupplyCoin := sdk.NewCoin(testDenom, initSupplyAmount)
+
+	tests := []struct {
+		inflation       types.Inflation
+		expectedBalance sdk.Int
+	}{
+		{types.NewInflation(testDenom, targetAccounts[0].Address.String(), sdk.NewDecWithPrec(1, 2)), sdk.NewInt(1576534791)},
```

Comment:
Perhaps I'm missing something, but expected values provided here shall be exaclty equal to  `supply * yearly_inlfation`. It kind of feels the numbers provided below do not match that condition.

Reply:
Please ignore my comment above, this test for rewards per block.